tests/cnf/core/network/sriov/tests/qinq.go

```diff
@@ -544,6 +580,112 @@ var _ = Describe("QinQ", Ordered, Label(tsparams.LabelQinQTestCases), ContinueOn
+			Expect(err).ToNot(HaveOccurred(), "Failed remove sriov configurations")
+
+			By("Creating a new instance of NMstate instance")
+			err = netnmstate.CreateNewNMStateAndWaitUntilItsRunning(7 * time.Minute)
+			Expect(err).ToNot(HaveOccurred(), "Failed to create NMState instance")
+
+			if sriovenv.IsMellanoxDevice(srIovInterfacesUnderTest[0], workerNodeList[0].Object.Name) {
+				configureSriovMlnxFirmwareOnWorkersAndWaitMCP(workerNodeList, srIovInterfacesUnderTest[0], true, 5)
+			}
+
+			By("Creating SR-IOV VFs via NMState")
+			err = netnmstate.ConfigureVFsAndWaitUntilItsConfigured(
+				configureNMStatePolicyName,
+				srIovInterfacesUnderTest[0],
+				NetConfig.WorkerLabelMap,
+				5,
+				netparam.DefaultTimeout)
+			Expect(err).ToNot(HaveOccurred(), "Failed to create VFs via NMState")
+
+			err = sriovenv.WaitUntilVfsCreated(workerNodeList, srIovInterfacesUnderTest[0], 5, netparam.DefaultTimeout)
+			Expect(err).ToNot(HaveOccurred(), "Expected number of VFs are not created")
+
+			By("Configure SR-IOV with flag ExternallyManaged true")
+			err = sriovenv.CreateSriovConfigurationWithExManaged(sriovAndResourceNameExManagedTrue,
+				srIovInterfacesUnderTest[0], true)
+			Expect(err).ToNot(HaveOccurred(),
+				"Failed to create sriov configuration with flag ExternallyManaged true")
+
+			By("Define and create sriov-network for the promiscuous client")
+			defineAndCreateSriovNetworks(srIovNetworkPromiscuous, srIovNetworkDot1AD, srIovNetworkDot1Q,
+				sriovAndResourceNameExManagedTrue)
+
+			promiscVFCommand = fmt.Sprintf("ethtool --set-priv-flags %s vf-true-promisc-support on",
+				srIovInterfacesUnderTest[0])
+			if sriovDeviceID == netparam.MlxDeviceID {
+				promiscVFCommand = fmt.Sprintf("ip link set %s promisc on",
+					srIovInterfacesUnderTest[0])
+			}
+
+			By(fmt.Sprintf("Enable VF promiscuous support on %s", srIovInterfacesUnderTest[0]))
+			output, err := cmd.RunCommandOnHostNetworkPod(workerNodeList[0].Definition.Name,
+				NetConfig.SriovOperatorNamespace, promiscVFCommand)
+			Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("Failed to run command on node %s", output))
+		})
+
+		It("Verify an 802.1ad QinQ tunneling between two containers with the VFs configured by NMState",
+			reportxml.ID("71681"), func() {
+				By("Define and create a container in promiscuous mode")
+				tcpDumpContainer := createPromiscuousClient(workerNodeList[0].Definition.Name,
+					tcpDumpNet1CMD)
+
+				By("Define and create a 802.1AD server container")
+				serverAnnotation := defineNetworkAnnotation(srIovNetworkDot1AD, nadCVLAN100, true)
+				serverPod := createServerTestPod(serverNameDot1ad, workerNodeList[0].Definition.Name, testCmdNet2,
+					serverAnnotation)
+
+				By("Define and create a 802.1AD client container")
+				clientAnnotation := defineNetworkAnnotation(srIovNetworkDot1AD, nadCVLAN100, false)
+				clientPod := createClientTestPod(clientNameDot1ad, workerNodeList[0].Definition.Name, clientAnnotation)
```

Comment:
Maybe I missed soemthing, but I don't see that client and server are using sriovNetwork(sriovAndResourceNameExManagedTrue) that created with exmanaged flag. Soemthing wrong here

Reply:
The sriovnetworks are recreated with the same name but are using the externally managed policy res name.

Reply:
I think, if you remove it nothing is changed
https://github.com/gkopels/eco-gotests/blob/qinq-add-nmstate-tc/tests/cnf/core/network/sriov/tests/qinq.go#L1145-L1151